sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallDialog.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.callautomation;
+
+import com.azure.core.util.Context;
+import com.azure.communication.callautomation.models.DialogStateResult;
+import com.azure.communication.callautomation.models.StartDialogOptions;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.rest.Response;
+
+/**
+ * CallDialog.
+ */
+public final class CallDialog {
+
+    /**
+     * Instance of CallDialogAsync.
+     */
+    private final CallDialogAsync callDialogAsync;
+
+    /**
+     * Constructor.
+     *
+     * @param callDialogAsync callDialogAsync
+     */
+    public CallDialog(CallDialogAsync callDialogAsync) {
```

Comment:
don't need public constructor as this is getting retrieved from getCallDialog method from callconnection